### ISSUE TITLE
Money Counted Strangely in Logic

### ISSUE THREAD

**Issue by jet082** [abyssonym/sai2_rando#3]
So in the logic you have lines like

`.def downjab 047f&sword&500_coins
`

Which means that downjab locations (like vanilla fireballs) are not in logic with having found the 1000 coin or 2000 coin bags, but only after finding the 500 coin bag. This isn't a problem, persay, but it is a bit unintuitive.

I'm not sure how else to handle this though without enabling grind since each of these money bags are loosely treated like "mushrooms" (to put it in alttpr terms).

Anyway, I figured I would raise the question, but I'll leave it to your judgement on the best way to address this - or to not do so at all.

**Comment by abyssonym** [comment]
That's a fair point. I can set it up so that higher-coin bags can also fulfill the "500 coins" condition. The change is simple, I'd just write something like
```
.def 1000_coins 0003|2000_coins
.def 500_coins 0002|1000_coins
```
etc.

**Comment by abyssonym** [comment]
I'd still use the 1000 coins condition for upjab, though, otherwise a single 500 coin pickup would unlock both techs, requiring up to 300 coins worth of grinding.

**Comment by Jigglysaint** [comment]
For me personally, I'd like to see the techniques shuffled into the item pool since I know you can obtain them via chests(though they look buggy).  It would also be cool to have chests added to the game, though that might be a bit too hard considering that part is background tiles, and part is a rather lengthy event code.  I have manged to add a chest to a new room, but it usually involves repointing the event data to make space for the new chest code.

**Comment by jet082** [comment]
> For me personally, I'd like to see the techniques shuffled into the item pool since I know you can obtain them via chests(though they look buggy). It would also be cool to have chests added to the game, though that might be a bit too hard considering that part is background tiles, and part is a rather lengthy event code. I have manged to add a chest to a new room, but it usually involves repointing the event data to make space for the new chest code.

If this is possible, I think it would be worth whatever buggy graphics are generated. It would make going to those far-off locations (especially in an open seed) an interesting gamble to make.

**Comment by Jigglysaint** [comment]
Looking at that now, what happens is instead of a message appearing, the hud glitches out for a moment.  That is because there are only enough item messages to go from values 00 to 2A.  Anything past that is reading other data as pointers.  If I can find that list, it could be possible to move things around and add new messages in.

**Comment by jet082** [comment]
> Looking at that now, what happens is instead of a message appearing, the hud glitches out for a moment. That is because there are only enough item messages to go from values 00 to 2A. Anything past that is reading other data as pointers. If I can find that list, it could be possible to move things around and add new messages in.

Several people in the community (LackAttack, etc) are planning a community race event thing to show this off, so it would ideally be nice to have this done before then.

edit: If this isn't possible though, nbd - the rando is already REALLY fun!

**Comment by Jigglysaint** [comment]
The good news is I manged to compile a simple .tbl file that locates all the text.  The downside is that the item names still don't show up.  It's possible they use a different font that has a different hex entry.  Dynamic text with hints for the seed could be a possibility with this.
[Sai2table .txt](https://github.com/abyssonym/sai2_rando/files/2880739/Sai2table.txt)

That's the .tbl file.  Rename it to .tbl to use it properly.

**Comment by Jigglysaint** [comment]
Okay great news.  I have found the location of the event items names and graphics.  I still currently do not know how the game syncs up the 7B chests with each text, but I know the location of the text tables and graphics that are used with them.  This means that it should be possible to put the techniques into an event chest, though I still need to find the values that sets the bits in ram for the techniques bought at the shops.

At offset 5C80, there is a table containing the information about the item graphics.  8 bytes for each object, and as expected, it terminates right before a new block of code.  If the table was repointed, and it can be repointed anywhere since each component uses a long instruction, it can be expanded.  The text table is all the way at 50000, and contains relative pointers.  The .tbl file can't read them so they must use some sort of other format for the text.  This too can be expanded or replaced.

**Comment by Jigglysaint** [comment]
Okay I created a table.  I was right, the item text is separate from the rest of the text.
[Sai2itemtext.txt](https://github.com/abyssonym/sai2_rando/files/2881181/Sai2itemtext.txt)
The text and the box it contains are all defined here.  There are a few things missing, like the numbers for the coins, but there is more than enough to work with here.
